IntelliJ Idea Glassfish plugin not working with 7.0.18

## Environment Details
* GlassFish Version (and build number): 7.0.18

----------

## Problem Description

IntelliJ cannot deploy applications for GlassFish and gives a `NoSuchMethodError` exception:

``
Error running admin process:
Message: java.lang.NoSuchMethodError: 'void com.sun.enterprise.admin.remote.RemoteResponseManager.<init>(java.io.InputStream, int, java.util.logging.Logger)' Stack trace: com.intellij.javaee.process.common.JavaeeProcessUtilException: java.lang.NoSuchMethodError: 'void com.sun.enterprise.admin.remote.RemoteResponseManager.<init>(java.io.InputStream, int, java.util.logging.Logger)' at com.intellij.javaee.process.common.MethodInvocator.invoke(MethodInvocator.java:47) at com.intellij.javaee.oss.process.JavaeeProcess.processRequest(JavaeeProcess.java:110) at com.intellij.javaee.oss.process.JavaeeProcess.run(JavaeeProcess.java:53) at com.intellij.javaee.oss.process.JavaeeProcess.main(JavaeeProcess.java:32) Caused by: java.lang.NoSuchMethodError: 'void com.sun.enterprise.admin.remote.RemoteResponseManager.<init>(java.io.InputStream, int, java.util.logging.Logger)' at com.intellij.javaee.oss.glassfish.agent.Glassfish31AgentBase.doParseResponse(Glassfish31AgentBase.java:21) at com.intellij.javaee.oss.glassfish.agent.Glassfish3AgentBase.parseResponse(Glassfish3AgentBase.java:195) at com.intellij.javaee.oss.glassfish.agent.Glassfish3AgentBase.invoke(Glassfish3AgentBase.java:144) at com.intellij.javaee.oss.glassfish.agent.Glassfish3AgentBase.doConnect(Glassfish3AgentBase.java:34) at com.intellij.javaee.oss.agent.SimpleAgentBase$1.doJob(SimpleAgentBase.java:24) at com.intellij.javaee.oss.agent.SimpleAgentBase$1.doJob(SimpleAgentBase.java:20) at com.intellij.javaee.oss.agent.SimpleAgentJob.perform(SimpleAgentJob.java:12) at com.intellij.javaee.oss.agent.SimpleAgentBase.connect(SimpleAgentBase.java:33) at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) at java.base/java.lang.reflect.Method.invoke(Method.java:580) at com.intellij.javaee.process.common.MethodInvocator.invoke(MethodInvocator.java:41) ... 3 more
```

## Steps to reproduce

1. Install the GlassFish plugin into IntelliJ
2. Install GlassFish 7.0.18 into IntelliJ
3. Create a sample Jakarta EE application
4. Run the application with GlassFish 7.0.18

IntelliJ is able to start GlassFish but fails to deploy the application with the `NoSuchMethodError` exception above.

## Impact of Issue

Not possible to use GlassFish 7.0.18 from IntelliJ.

[//]: # "Please choose one each Type, Component, and Priority label"
Type: Bug
Component: Deployment, IDE support
Priority: High

--- a/nucleus/admin/util/src/main/java/com/sun/enterprise/admin/remote/RemoteResponseManager.java
+++ b/nucleus/admin/util/src/main/java/com/sun/enterprise/admin/remote/RemoteResponseManager.java
@@ -52,6 +52,21 @@ public class RemoteResponseManager implements ResponseManager {
     private final String response;
     private Map<String, String> mainAtts = Collections.emptyMap();
 
+    /**
+     * Creates the object with the default charset by delegating to the {@link #RemoteResponseManager(java.io.InputStream, java.nio.charset.Charset, int, java.util.logging.Logger)} constructor.
+     * @param in
+     * @param code
+     * @param logger
+     * @throws RemoteException
+     * @throws IOException
+     * @deprecated It is encouraged to use the {@link #RemoteResponseManager(java.io.InputStream, java.nio.charset.Charset, int, java.util.logging.Logger)} constructor with explicit charset. This constructor is kept only for backwards compatibility with the IntelliJ Idea plugin.
+     */
+    @Deprecated(since = "7.0.18")
+    public RemoteResponseManager(InputStream in, int code, Logger logger)
+        throws RemoteException, IOException {
+        this(in, Charset.defaultCharset(), code, logger);
+    }
+
     public RemoteResponseManager(InputStream in, Charset charset, int code, Logger logger)
         throws RemoteException, IOException {
         this.code = code;

